Add wallet mode

nmoskaleva opened the pull request:
- Wallet mode added.
- Loading state: added to remedy the flashing login screen after successful authentication.
- Homepage link in Header logo added.
- `@criipto/verify-react` updated to the latest version.
-  Newer version of `@types/react` is not compatible with current `@material-tailwind/react`: fixed by locking both versions.

mickhansen commented on src/Components/Modal.tsx:
```diff
@@ -42,6 +42,29 @@ export default function Modal() {
+        sessionStorage.removeItem('wallet');
+      } else {
+        params.set('wallet', 'true');
+        sessionStorage.setItem('wallet', 'true');
```
Why `sessionStorage`? We don't appear to be using it for the other parameters.

nmoskaleva replied:
I thought it makes sense to not have to switch to the wallet mode every time. 
Maybe this is just my perception, since I was running test logins so many times, and having to switch to the wallet mode every time seemed wrong.
Let's remove it them.

mickhansen replied:
@nmoskaleva Ah right! Well in that case we could perhaps have the logout action include the query params for wallet?

mickhansen replied:
(`redirect_uri` can contain query params)

nmoskaleva replied:
Ah that's much better! Thanks

nmoskaleva replied:
@mickhansen please correct me if I'm wrong, but it seems much easier to use `sessionStorage` here, since we'd have to keep track of some kind of wallet mode state anyway to handle logouts. Are there any disadvantages to it?

mickhansen replied:
@nmoskaleva The disadvantage is having to reason about session storage state rather than just looking at query params.

We only have 3-4 routes, so carrying around query params seems pretty reasonable to me?

nmoskaleva replied:
Gotcha. I thought having to add a wallet state to handle logouts might be more complicated than having `sessionStorage`. Thanks for explaining, I'll make changes.

mickhansen replied:
Should be straight forward, but let me know if it's too complicated!

nmoskaleva commented:
Updates: 
- Routing changes: changed redirect URL to`/login/callback` and added a `Loading` page that handles redirects depending on the `claims`. _Seems to have resolved all the intermittent routing issues I was having._
- Desktop UX fix: Homepage now includes the wallet mode toggle and an updated Login button. 
* Still using `sessionStorage` to keep track of wallet mode for now. 
* Redirect URLs in the application have to be updated to include `.../login/callback`.

mickhansen commented on src/Hooks/useWallet.tsx:
```diff
@@ -0,0 +1,21 @@
+const useWalletLogin = () => {
+  const { loginWithRedirect } = useCriiptoVerify();
+
+  const handleWalletLogin = () => {
```
Probably want `useMemo`

mickhansen commented on src/Hooks/useWallet.tsx:
```diff
@@ -0,0 +1,21 @@
+import { useSearchParams } from 'react-router-dom';
+import { useCriiptoVerify } from '@criipto/verify-react';
+
+const useWalletMode = () => {
```
Set return type to boolean

mickhansen commented on src/index.tsx:
```diff
@@ -19,7 +20,7 @@ function CriiptoVerifyProviderWrapper() {
       domain={domain}
       clientID="urn:demos:cool-energy-react"
-      redirectUri={window.location.origin + '/login'}
+      redirectUri={window.location.origin + '/login/callback'}
```
may want to include `wallet` in the query here if wallet mode

mickhansen commented on src/Components/Loading.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const Loading = () => {
+  const navigate = useNavigate();
+  const { claims } = useCriiptoVerify();
```
I think actually that `useCriiptoVerify` returns isLoading and other values you can use, then we might be able to keep the current Routes logic.

nmoskaleva replied:
It does return `isLoading`, I'm even using it :) 
So keep redirecting to the Dashboard directly, without `/login/callback` then?

mickhansen replied:
Up to you, a seperate screen might be easier to reason about

mickhansen commented on src/App.tsx:
```diff
@@ -5,19 +5,24 @@ import Dashboard from './Components/Dashboard';
     <>
-      <Route path="login" element={<Login />} />
+      <Route path="/login" element={<Login />} />
+      <Route path="/login/callback" element={<Loading />} />
```
Having a `Loading  element in `/login/callback` seems a tad odd? Perhaps the component should be called LoginCallback?

nmoskaleva commented:
- Routing issues are fixed after taking `isLoading` into account.
- Remove a dedicated component for `/login/callback` route that is now always redirecting to the Dashboard or Login. 
- Add a `Loading` component

mickhansen commented on src/Components/Header.tsx:
```diff
@@ -21,13 +24,15 @@ function Header(props: Props) {
-          className="h-10 w-[220px]"
-        />
-      </div>
+      <a href="/">
```
include wallet mode

nmoskaleva replied:
Done
